src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.cs

```diff
@@ -125,9 +125,45 @@ public static partial class JsonSerializer
                 ThrowHelper.ReThrowWithPath(e, state.PropertyPath);
             }
 
+            // Ensure any trailing whitespace or comments (if allowed) are cleaned up
+            ConsumeTrailingWhitespaceAndComments(ref reader, options);
+
             return;
         }
 
+        private static void ConsumeTrailingWhitespaceAndComments(ref Utf8JsonReader reader, JsonSerializerOptions options)
+        {
+            try
+            {
+                while (reader.Read())
+                {
+                    switch (reader.TokenType)
+                    {
+                        case JsonTokenType.None:
```

Comment:
I don't know if we will ever hit this `None` case here. If you run code coverage, do you see this case being hit by any of the tests?

Reply:
It doesn't look like any of this code is being hit. To me, that partially indicates all these cases aren't necessary.
![image](https://user-images.githubusercontent.com/6527137/57496165-61974b00-7286-11e9-8de2-9c6b86e8d835.png)

Reply:
I was borrowing other code, which based on your other suggestion appear to be at a lower level than what was needed.